pkg/querier/queryrange/queryrange.proto

```diff
@@ -160,14 +161,27 @@ message QuantileSketchResponse {
 }
 
 message QueryResponse {
+  google.rpc.Status status = 1;
   oneof response {
-    LokiSeriesResponse series = 1;
-    LokiLabelNamesResponse labels = 2;
-    IndexStatsResponse stats = 3;
-    LokiPromResponse prom = 4;
-    LokiResponse streams = 5;
-    VolumeResponse volume = 6;
-    TopKSketchesResponse topkSketches = 7;
-    QuantileSketchResponse quantileSketches = 8;
+    LokiSeriesResponse series = 2;
+    LokiLabelNamesResponse labels = 3;
+    IndexStatsResponse stats = 4;
+    LokiPromResponse prom = 5;
+    LokiResponse streams = 6;
+    VolumeResponse volume = 7;
+    TopKSketchesResponse topkSketches = 8;
+    QuantileSketchResponse quantileSketches = 9;
   }
 }
+
+message QueryRequest {
+  oneof request {
+    LokiSeriesRequest series = 1;
+    logproto.LabelRequest labels = 2;
+    logproto.IndexStatsRequest stats = 3;
+    LokiInstantRequest instant = 4;
+    LokiRequest streams = 5;
+    logproto.VolumeRequest volume = 6;
+  }
+  map<string, string> metadata = 7 [(gogoproto.nullable) = false];
```

Comment:
After some issues I think all requests and responses should have a `SetMetadata` and `GetMetadata` method.